modules/network/samba.c

```diff
@@ -51,33 +51,36 @@ void
 scan_samba_usershares(void)
 {
     FILE *usershare_list;
-    
-    if ((usershare_list = popen("net usershare list", "r"))) {
-        char buffer[512];
-        
+    gchar *out, *err;
+    gboolean spawned;
+    int status;
+    gchar *p, *next_nl;
+    gchar *usershare, *cmdline;
+    gsize length;
+
+    spawned = g_spawn_command_line_sync("net usershare list",
+            &out, &err, &status, NULL);
```

Comment:
If `stderr` is not required, you can pass `NULL` instead of `&err`; then there's no need to `free()` it below.